### PR TITLE
Cli: add subcommand to withdraw from vote account

### LINKED ISSUE #8095: cli: no ability to withdraw funds from a vote account
As a vote account collects rewards there's no easy way to withdraw those funds, as `VoteInstruction::Withdraw` is not plumbed through the solana cli.

At the same time, consider whether the fact that `VoteInstruction::Withdraw` permits the user to create a non-rent-exempt vote account would be a problem:  https://github.com/solana-labs/solana/blob/927f272f0e22131ce57713a9d7e9792de16d0ad1/programs/vote/src/vote_state.rs#L500-L509

### PR DESCRIPTION
**Pull request by sunnygleason**
#### Problem

The product's command line doesn't offer the ability to transfer rewards easily.

#### Summary of Changes

Add a vote account withdraw command.

Fixes #8095

### CODE REVIEW

File: cli/src/vote.rs [hunk 1]
```diff
@@ -276,6 +317,35 @@ pub fn parse_vote_get_account_command(
...
+pub fn parse_vote_account_withdraw_command(
+    matches: &ArgMatches<'_>,
+    default_signer_path: &str,
+    wallet_manager: Option<&Arc<RemoteWalletManager>>,
+) -> Result<CliCommandInfo, CliError> {
+    let vote_account_pubkey = pubkey_of(matches, "vote_account_pubkey").unwrap();
+    let withdrawer_account_pubkey = pubkey_of(matches, "withdrawer_account_pubkey").unwrap();
+    let lamports: u64 = matches.value_of("lamports").unwrap().parse().unwrap();
+    let to_account_pubkey = pubkey_of(matches, "to_account_pubkey").unwrap();
+
```
Comment (sunnygleason) [inline cli/src/vote.rs]: I don't have an account to easily test this -- feedback would be super useful from someone who can actually try this out...

**Comment by mvines** [comment]
@t-nelson - hey can you please pick up this PR from @sunnygleason.   There's a little bit more cli plumbing to do, and you're familiar with all of it so hopefully should be quick

File: cli/src/vote.rs (hunk 1 above)
Comment (CriesofCarrots) [reply cli/src/vote.rs]: ✔️ Integration test added!

File: cli/tests/vote.rs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        authorized_voter: None,
+        authorized_withdrawer: Some(config.signers[0].pubkey()),
+        commission: 0,
+    };
+    process_command(&config).unwrap();
+    let vote_account = rpc_client
+        .get_account(&vote_account_keypair.pubkey())
+        .unwrap();
+    let vote_state: VoteStateVersions = vote_account.state().unwrap();
+    let authorized_withdrawer = match vote_state {
```
Comment (t-nelson) [inline cli/tests/vote.rs]: You can shortcut this match with...
`vote_state.convert_to_current().authorized_withdrawer`

File: cli/tests/vote.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        vote_account_pubkey,
+        new_authorized_pubkey: withdraw_authority.pubkey(),
+        vote_authorize: VoteAuthorize::Withdrawer,
+    };
+    process_command(&config).unwrap();
+    let vote_account = rpc_client
+        .get_account(&vote_account_keypair.pubkey())
+        .unwrap();
+    let vote_state: VoteStateVersions = vote_account.state().unwrap();
+    let authorized_withdrawer = match vote_state {
```
Comment (t-nelson) [inline cli/tests/vote.rs]: ditto

File: cli/tests/vote.rs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+use solana_faucet::faucet::run_local_faucet;
+use solana_sdk::{
+    account_utils::StateMut,
+    pubkey::Pubkey,
+    signature::{Keypair, Signer},
+};
+use solana_vote_program::vote_state::{VoteAuthorize, VoteState, VoteStateVersions};
+use std::{fs::remove_dir_all, sync::mpsc::channel, thread::sleep, time::Duration};
+
+fn check_balance(expected_balance: u64, client: &RpcClient, pubkey: &Pubkey) {
```
Comment (t-nelson) [inline cli/tests/vote.rs]: I think this function is in every file in this dir :laughing:

File: cli/tests/vote.rs (hunk 2 above)
Comment (CriesofCarrots) [reply cli/tests/vote.rs]: Oh, thank you! I missed that.

File: cli/tests/vote.rs (hunk 4 above)
Comment (CriesofCarrots) [reply cli/tests/vote.rs]: 🙈

**Comment by CriesofCarrots** [comment]
@mvines , do you want to review this?
Does this need to go in v1.0?

File: cli/src/stake.rs [hunk 5]
```diff
@@ -34,13 +34,13 @@ use std::{ops::Deref, sync::Arc};
 pub const STAKE_AUTHORITY_ARG: ArgConstant<'static> = ArgConstant {
     name: "stake_authority",
     long: "stake-authority",
-    help: "Public key of authorized staker (defaults to cli config pubkey)",
+    help: "Authorized staker (defaults to cli config keypair)",
```
Comment (mvines) [inline cli/src/stake.rs]: nit: make our manual default look like what clap generates
```suggestion
    help: "Authorized staker [default: cli config keypair]",
```

File: cli/src/stake.rs [hunk 6]
```diff
@@ -34,13 +34,13 @@ use std::{ops::Deref, sync::Arc};
...
 };
 
 pub const WITHDRAW_AUTHORITY_ARG: ArgConstant<'static> = ArgConstant {
     name: "withdraw_authority",
     long: "withdraw-authority",
-    help: "Public key of authorized withdrawer (defaults to cli config pubkey)",
+    help: "Authorized withdrawer (defaults to cli config keypair)",
```
Comment (mvines) [inline cli/src/stake.rs]:
```suggestion
    help: "Authorized withdrawer [default: cli config keypair]",
```

**Comment by mvines** [comment]
> Does this need to go in v1.0?

It doesn't need to but for low-risk cli additions like this, I'm all for it.

File: cli/src/stake.rs (hunk 5 above)
Comment (CriesofCarrots) [reply cli/src/stake.rs]: I bet there are a few other uses of that format. I'll update all.

**Pull request merged**